### PR TITLE
Handle responses using virtus models

### PR DESCRIPTION
**Pull request by jdks**
Previously API responses weren't being handled. This change wraps the response object in the appropriate virtus models in order to set up basic accessors. Keys in the JSON response are converted from camel case to snake case.

### CODE REVIEW

File: lib/geonames/places/response.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+require 'geonames/places/status'
+require 'geonames/places/record'
+
+module Geonames
+  module Places
+    class Response
+      include Virtus.model
+
+      attribute :status, Status
+
```
Comment (gavinlaking) [inline lib/geonames/places/response.rb]: Is there a missing attribute here?

File: lib/geonames/places/wrapper.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+          end
+        end
+      end
+
+      def normalized_keys
+        @data.keys.lazy.map(&(method :normalize))
+      end
+
+      def normalize(key)
+        key.gsub(/(\p{Lu})\p{Ll}/) { |match| "_" + match[/\p{Lu}\p{Ll}/].downcase }
```
Comment (gavinlaking) [inline lib/geonames/places/wrapper.rb]: I'd recommend extracting this regular expression to a method, so we can use a name to describe what it does.

File: spec/geonames/places/response_spec.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+module Geonames
+  module Places
+    RSpec.describe Response do
```
Comment (gavinlaking) [inline spec/geonames/places/response_spec.rb]: Perhaps we can specify the attributes we expect this model will have here?

File: spec/geonames/places/status_spec.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+module Geonames
+  module Places
+    RSpec.describe Status do
```
Comment (gavinlaking) [inline spec/geonames/places/status_spec.rb]: Perhaps we can specify the attributes we expect this model will have here?

File: spec/geonames/places/wrapper_spec.rb [hunk 5]
```diff
@@ -0,0 +1,6 @@
+module Geonames
+  module Places
+    RSpec.describe Wrapper do
```
Comment (gavinlaking) [inline spec/geonames/places/wrapper_spec.rb]: Perhaps we can specify the attributes we expect this model will have here?

File: lib/geonames/places/response.rb (hunk 1 above)
Comment (jdks) [inline lib/geonames/places/response.rb]: Not for any example responses I've come across. Here's one:

```json
{
  "status": {
    "message": "sibling service only supports feature class A",
    "value": 21
  }
}
```

**Pull request closed without merging**